### PR TITLE
[ci skip][WIP] Freecad: switch to Qt5 and python3

### PR DESCRIPTION
**Pull request by yopito**
* huge build, so [ci skip]
* personal native build : OK
* personal native test: OK ("`FreeCAD -t 0`")
* unable to test crossbuild these days:
  * aarch64: "qt5-5.11.3_5: broken, unresolvable shlib `libstdc++.so.6'"
  * i686: "cairo-1.16.0_1: broken, unresolvable shlib `ld-linux.so.2'"

@Piraty, @karl-nilsson, @luzpaz: help and advices welcome.

### CODE REVIEW

**Comment by yopito** [comment]
forgot revdump on FreeCAD template... here it is

**Comment by jnbr** [comment]
[ci skip] has to be in the commit message of HEAD to be effective.

**Comment by yopito** [comment]
@jnbr: added in last commit, thanks for pointing this

**Comment by Johnnynator** [comment]
shiboken2 fails for me with:
```
[ 48%] Building CXX object ApiExtractor/CMakeFiles/apiextractor.dir/reporthandler.cpp.o
/builddir/pyside-setup-everywhere-src-5.11.2/sources/shiboken2/libshiboken/sbknumpyarrayconverter.cpp:46:10: fatal error: numpy/arrayobject.h: No such file or directory
 #include <numpy/arrayobject.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libshiboken/CMakeFiles/libshiboken.dir/build.make:297: libshiboken/CMakeFiles/libshiboken.dir/sbknumpyarrayconverter.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 50%] Building CXX object ApiExtractor/CMakeFiles/apiextractor.dir/typeparser.cpp.o
make[1]: *** [CMakeFiles/Makefile2:231: libshiboken/CMakeFiles/libshiboken.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

File: srcpkgs/python-pivy/template [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-
-# depends on script coin-config
-nocross=yes
-
-short_desc="Python binding for the high-level 3D graphics library Coin"
+hostmakedepends="python3-devel swig"
+makedepends="python3-devel coin3-devel qt5-devel"
+short_desc="Python3 binding for the high-level 3D graphics library Coin"
 maintainer="yopito <pierre.bourgin@free.fr>"
 license="BSD"
```
Comment (maxice8) [inline srcpkgs/python-pivy/template]: needs to be an SPDX short-identifier license.

**Comment by maxice8** [comment]
Please remove python-pivy and instead add python3-pivy

**Comment by yopito** [comment]
@Johnnynator : rather strange. what's your build/arch ? could you paste your packaging log of shiboken2 into a gist ? your build env is clean ?

**Comment by karl-nilsson** [comment]
@yopito I built and tested your packages (x64, glibc), and everything seems to work, including the test suite.

One very minor bug: in the "About FreeCAD" dialog, 2 of the fields (revision number & release date) display "Unknown". I dug into the code, and it looks like the cmake script runs a python script (src/Tools/SubWCRev.py) to fill those values in by querying git/bzr/svn. Probably the easiest solution is to patch src/Build/CMakeLists.txt and hardcode the desired values. 
As this is a very minor bug, I'll leave it up to your judgement.

**Comment by yopito** [comment]
@karl-nilsson : thanks for your return. I will give a try to provide the release date and SCM revision.

x86_64/musl status: native build succedeed, but runtime fails.  
SIGSEGV, possibly on libQt5WebKit.so.5. I'm not familiar with this kind of debug.  
Can someone have a look on the gdb backtrace [1] and confirm sefault is related to Qt lib ? Thanks.
[1] https://gist.github.com/yopito/3408054baf3626f6a167d6521dfa2d50

**Comment by yopito** [comment]
x86_64/musl status: SIGSEGV, probably related with libQt5WebKit.so.5
updated gdb traceback with more valuable info [1].
@pullmoll : could you have a quick look on this traceback if you have some hint about it ? many thanks.

[1] https://gist.github.com/yopito/3408054baf3626f6a167d6521dfa2d50

**Comment by pullmoll** [comment]
@yopito that could again be a problem with musl libc's tight stack size. You could perhaps try `chelf` on /usr/lib/freecad/bin/FreeCAD.

**Comment by yopito** [comment]
@pullmoll : thanks for your reply.

x86_64/musl: Tried setting raising stack size with `chelf`: still SIGSEV, even at 80 MB ...  
The FreeCAD "start" workbench (auto-loaded by default) is the problem. It's embedding a Qt web browser. Without this workbench, start is fine and FreeCAD's internal tests are partially OK.

**Comment by pullmoll** [comment]
@yopito Perhaps we can find what part of QtWebKit's javascript compiler is triggering the (seemingly) stack corruption.  The function `JSC::JSGlobalObject::init(JSC::VM&)` is to be inspected and it is in `qtwebkit-everywhere-src-5.212/Source/JavaScriptCore/runtime/JSGlobalObject.cpp` line 258 ff.
I'll have a look at it.

**Comment by yopito** [comment]
Fail to fix runtime SIGSEV on x86_64/musl.   
Also had a look to FreeCAD-current: Start WorkBench has been completely rewritten, so useless to spend to fix the 0.17 one ...  
so I close this PR.  
I will open another one with "basic" updates for FreeCAD 0.17, remaining with qt4 and py2, that is fine for both glibc and musl.  
Thanks for your inputs for this one, I will reuse it to track next FreeCAD release.

**Pull request closed without merging**